### PR TITLE
core: Support wepb in image tag

### PR DESCRIPTION
**Pull request by mmzeeman**
### Description

See #2884 

This PR allows the image tag to produce webp formatted images. See: https://developers.google.com/speed/webp

Modern image formats are often recommended by Lighthouse. This makes
it possible to easily enable it.

### Checklist

- [x] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

File: apps/zotonic_core/src/support/z_media_preview.erl [hunk 1]
```diff
@@ -387,13 +387,19 @@ out_mime(Mime, Options, Context) ->
 %% @doc Return the preferred mime type of the image generated by resizing an image of a certain type and size.
 -spec out_mime( InMime :: binary(), list() ) -> { OutMime :: binary(), Extension :: string()}.
 out_mime(Mime, Options) ->
-    out_mime1(get_lossless_value(Options), Mime).
+    Mime1 = case proplists:get_value(webp, Options, false) of
```
Comment (mworrell) [inline apps/zotonic_core/src/support/z_media_preview.erl:390]: I wonder if we should make this a "format" option to enable more formats in the future?

**Review by mworrell — APPROVED** [review]

File: apps/zotonic_core/src/support/z_media_preview.erl (hunk 1 above)
Comment (mmzeeman) [reply apps/zotonic_core/src/support/z_media_preview.erl:390]: Ah.. yes... that would be more future proof indeed.

Comment (mworrell) [reply apps/zotonic_core/src/support/z_media_preview.erl:390]: We can map an extension to mime format, using the `[Mime|_] = mimetypes:ext_to_mimes(<<"webp">>)` or just use a list of supported formats of which we know it works, with a fallback to webp or jpeg.

I think the list of known formats might be best.

Comment (mmzeeman) [reply apps/zotonic_core/src/support/z_media_preview.erl]: Right now the "lossless" option decides if the extension will be png or jpg depending on the setting. Right now I added webp to this in a backward compatible way. Modern image formats have the possibility to choose between lossy and lossless compression. We might want to change this configuration setting for `Z 1.1`.

**Pull request merged**